compute/reservations/consume_specific_reservation.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START compute_consume_single_project_reservation]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "cloud.google.com/go/compute/apiv1/computepb"
+	"google.golang.org/protobuf/proto"
+)
+
+// Creates the reservation from given template in particular zone
+func createSpecificConsumableReservation(w io.Writer, projectID, zone, reservationName, sourceTemplate string) error {
```

Comment:
issue: same as previous. If this function is required as a snippet, it should be in its own code file.

See: https://googlecloudplatform.github.io/samples-style-guide/#one-per-file

Also, it looks very similar to `createConsumableReservation()`. Is there any code that makes it unique compared to the other function.

Reply:
Thanks for pointing out, but here is the same situation. In [gcloud example](https://cloud.google.com/compute/docs/instances/reservations-consume#consuming_a_specific_single-project_reservation) it is separated into 2 steps, but this is still a single sample. 

> Is there any code that makes it unique compared to the other function.

Yes, there is additional field, which makes the difference to consuming 
`SpecificReservationRequired: proto.Bool(true)`

I can make all actions being made in a single function, will it be better?